libs/datamodel/core/tests/types/mysql_native_types.rs

```diff
@@ -410,3 +410,355 @@ fn should_fail_on_native_type_numeric_when_scale_is_bigger_than_precision() {
         ast::Span::new(281, 311),
     ));
 }
+
+#[test]
+fn should_fail_on_native_type_text_with_id_attribute() {
+    let dml = r#"
+        datasource db {
+          provider = "mysql"
+          url = "mysql://"
+        }
+
+        generator js {
+            provider = "prisma-client-js"
+            previewFeatures = ["nativeTypes"]
+        }
+
+        model Blog {
+            bigInt String @db.Text @id
+        }
+
+        model User {
+            firstname String @db.Text
+            lastname  String @db.Text
+            @@id([firstname, lastname])
+        }
+    "#;
+
+    let error = parse_error(dml);
+
+    error.assert_length(2);
+
+    error.assert_is_at(
+        0,
+        DatamodelError::new_connector_error(
+            "Native type Text can not be used in an ID field in MySQL.",
+            ast::Span::new(247, 274),
+        ),
+    );
+    error.assert_is_at(
+        1,
+        DatamodelError::new_connector_error(
+            "Native type Text can not be used in an ID field in MySQL.",
+            ast::Span::new(293, 431),
+        ),
+    );
+}
+
+#[test]
+fn should_fail_on_native_type_long_text_with_id_attribute() {
+    let dml = r#"
+        datasource db {
+          provider = "mysql"
+          url = "mysql://"
+        }
+
+        generator js {
+            provider = "prisma-client-js"
+            previewFeatures = ["nativeTypes"]
+        }
+
+        model Blog {
+            bigInt String @db.LongText @id
+        }
+
+        model User {
+            firstname String @db.LongText
+            lastname  String @db.LongText
+            @@id([firstname, lastname])
+        }
+    "#;
+
+    let error = parse_error(dml);
+
+    error.assert_length(2);
+
+    error.assert_is_at(
+        0,
+        DatamodelError::new_connector_error(
+            "Native type LongText can not be used in an ID field in MySQL.",
+            ast::Span::new(247, 278),
+        ),
+    );
+    error.assert_is_at(
+        1,
+        DatamodelError::new_connector_error(
+            "Native type LongText can not be used in an ID field in MySQL.",
+            ast::Span::new(297, 443),
+        ),
+    );
```

Comment:
I think you can collapse all those very similar test cases into one. You could introduce a Vector of the native types that need to be tested. Then you loop over those and create the schemas dynamically and execute the test portion. This is probably a bit of work to get the error assertion right but you will figure it out :-)